homeassistant/components/freebox/router.py

```diff
@@ -50,49 +43,33 @@ async def get_api(hass: HomeAssistant, host: str) -> Freepybox:
 class FreeboxRouter:
     """Representation of a Freebox router."""
 
-    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        api: Freepybox,
+        freebox_config: Mapping[str, Any],
+    ) -> None:
         """Initialize a Freebox router."""
         self.hass = hass
         self._entry = entry
         self._host = entry.data[CONF_HOST]
         self._port = entry.data[CONF_PORT]
 
-        self._api: Freepybox = None
-        self.name = None
-        self.mac = None
-        self._sw_v = None
-        self._attrs = {}
+        self._api: Freepybox = api
+        self.name: str = freebox_config["model_info"]["pretty_name"]
+        self.mac: str = freebox_config["mac"]
+        self._sw_v: str = freebox_config["firmware_version"]
+        self._attrs: dict[str, Any] = {}
 
         self.devices: dict[str, dict[str, Any]] = {}
         self.disks: dict[int, dict[str, Any]] = {}
         self.sensors_temperature: dict[str, int] = {}
         self.sensors_connection: dict[str, float] = {}
         self.call_list: list[dict[str, Any]] = []
 
-        self._unsub_dispatcher = None
-        self.listeners = []
-
-    async def setup(self) -> None:
-        """Set up a Freebox router."""
-        self._api = await get_api(self.hass, self._host)
-
-        try:
-            await self._api.open(self._host, self._port)
-        except HttpRequestError:
-            _LOGGER.exception("Failed to connect to Freebox")
-            return ConfigEntryNotReady
-
-        # System
-        fbx_config = await self._api.system.get_config()
-        self.mac = fbx_config["mac"]
-        self.name = fbx_config["model_info"]["pretty_name"]
-        self._sw_v = fbx_config["firmware_version"]
-
-        # Devices & sensors
-        await self.update_all()
-        self._unsub_dispatcher = async_track_time_interval(
-            self.hass, self.update_all, SCAN_INTERVAL
-        )
+        self._unsub_dispatcher: CALLBACK_TYPE | None = None
+        self.listeners: list[CALLBACK_TYPE] = []
```

Comment:
Is this used anywhere?

Reply:
Listeners is used in the `device_tracker` platform:
https://github.com/home-assistant/core/blob/f82183f0e3409edb0815476055202d39aafd645c/homeassistant/components/freebox/device_tracker.py#L30-L32

Reply:
But I don't see it used anywhere. It seems to be populated but never used.

Reply:
We could use `entry.async_on_unload` there instead.